pallets/settlement/src/lib.rs

```diff
@@ -2031,6 +2200,102 @@ impl<T: Config> Module<T> {
         Ok(())
     }
 
+    /// Affirms the instruction as a mediator.
+    fn base_affirm_instruction_as_mediator(
+        origin: T::RuntimeOrigin,
+        instruction_id: InstructionId,
+        expiry: Option<T::Moment>,
+    ) -> DispatchResult {
+        let (caller_did, _, instruction) =
+            Self::ensure_origin_perm_and_instruction_validity(origin, instruction_id, false)?;
+
+        // Verifies if the caller is a mediator
+        let mediator_affirmation_status =
+            InstructionMediatorsAffirmations::<T>::get(instruction_id, caller_did);
+        ensure!(
+            mediator_affirmation_status != MediatorAffirmationStatus::Unknown,
+            Error::<T>::CallerIsNotAMediator
+        );
+
+        // Verifies if the expiry date is in the future
+        let timestamp = <pallet_timestamp::Pallet<T>>::get();
+        if let Some(expiry) = expiry {
+            ensure!(expiry > timestamp, Error::<T>::InvalidExpiryDate);
+        }
+
+        // Updates the mediator's affirmation status to affirmed
+        InstructionMediatorsAffirmations::<T>::insert(
+            instruction_id,
+            caller_did,
+            MediatorAffirmationStatus::Affirmed { expiry, timestamp },
+        );
+        // If the mediator is not reaffirming the instruction, the number of pending affirmation must be updated
+        if MediatorAffirmationStatus::Pending == mediator_affirmation_status {
+            InstructionAffirmsPending::mutate(instruction_id, |n| *n = n.saturating_sub(1));
+        }
+        // If all affirmations have been received, the instruction will be scheduled for the next block
+        let n_pending_affirmations = InstructionAffirmsPending::get(instruction_id);
+        if n_pending_affirmations == 0
+            && instruction.settlement_type == SettlementType::SettleOnAffirmation
+        {
+            let instruction_asset_count = Self::get_instruction_asset_count(&instruction_id);
+            let weight_limit = Self::execute_scheduled_instruction_weight_limit(
+                instruction_asset_count.fungible(),
+                instruction_asset_count.non_fungible(),
+                instruction_asset_count.off_chain(),
+            );
+            Self::maybe_schedule_instruction(n_pending_affirmations, instruction_id, weight_limit);
+        }
```

Comment:
We do, but it's inside `maybe_schedule_instruction`, which means we would always have to call `Self::get_instruction_asset_count(&instruction_id)`. We can probably change that function so that we can get the `weight_limit` there, but this PR is already too large.